File: servers/dropwizard/src/main/java/com/slabstech/revive/server/dropwizard/retrofit/guide/GithubUserService.java
package com.slabstech.revive.server.dropwizard.retrofit.guide;

import com.slabstech.revive.server.dropwizard.retrofit.guide.GithubUser;

import java.util.List;

public interface GithubUserService {
    @GET("/users")
    public Call<List<GithubUser>> getUsers(
            @Query("per_page") int per_page,
            @Query("page") int page);

    @GET("/users/{username}")
    public Call<GithubUser> getUser(@Path("username") String username);
}

File: servers/dropwizard/src/main/java/com/slabstech/revive/server/dropwizard/retrofit/guide/GithubUserService.kt
package com.slabstech.revive.server.dropwizard.retrofit.guide

interface GithubUserService {
    @GET("/users")
    fun getUsers(
        @Query("per_page") per_page: Int,
        @Query("page") page: Int
    ): Call<List<GithubUser?>?>?

    @GET("/users/{username}")
    fun getUser(@Path("username") username: String?): Call<GithubUser?>?
}